Sources/Actions/Activate.php

```diff
@@ -94,36 +93,37 @@ class Activate implements ActionInterface
 	 */
 	public function execute(): void
 	{
-		if (empty($_REQUEST['u']) && empty($_POST['user']))
-		{
-			if (empty(Config::$modSettings['registration_method']) || Config::$modSettings['registration_method'] == '3')
+		if (empty($_REQUEST['u']) && empty($_POST['user'])) {
+			if (empty(Config::$modSettings['registration_method']) || Config::$modSettings['registration_method'] == '3') {
 				ErrorHandler::fatalLang('no_access', false);
+			}
 
 			Utils::$context['member_id'] = 0;
 			Utils::$context['sub_template'] = 'resend';
 			Utils::$context['page_title'] = Lang::$txt['invalid_activation_resend'];
 			Utils::$context['can_activate'] = empty(Config::$modSettings['registration_method']) || Config::$modSettings['registration_method'] == '1';
-			Utils::$context['default_username'] = isset($_GET['user']) ? $_GET['user'] : '';
+			Utils::$context['default_username'] = $_GET['user'] ?? '';
 
 			return;
 		}
 
 		// Get the code from the database...
-		$request = Db::$db->query('', '
+		$request = Db::$db->query(
+			'',
+			'
```

Comment:
This one's funny. Even though it looks strange, it's consistent with other params on their own lines.

Reply:
Well its a bc breakage.  But we could introduce a identifier_query() and drop the first param off this one.  Considering like 95% of our queries don't need this.

Reply:
> Well its a bc breakage. But we could introduce a identifierQuery() and drop the first param off this one. Considering like 95% of our queries don't need this.

That's a good idea. We'd just need to switch which method the backward compatibility entry in Utils::$smcFunc pointed to, and we'd be good to go.

Reply:
Hm. Here's another option that would introduce fewer complications for backward compatibility.

```
		$request = Db::$db->query(
			'',
			'SELECT id_member, validation_code, member_name, real_name, email_address, is_activated, passwd, lngfile
			FROM {db_prefix}members' . (empty($_REQUEST['u']) ? '
			WHERE member_name = {string:email_address} OR email_address = {string:email_address}' : '
			WHERE id_member = {int:id_member}') . '
			LIMIT 1',
			[
				'id_member' => isset($_REQUEST['u']) ? (int) $_REQUEST['u'] : 0,
				'email_address' => $_POST['user'] ?? '',
			],
		);
```

Taking out the extra line break at the start of the query string makes it look a lot less weird in my opinion.

Reply:
Does help but still looks odd.  If this is easier then go this way.  Maybe we introduce that in 3.1 or another future release.

Reply:
> Well its a bc breakage

Not is not. Bots never do bc breaks unless specifically told to.

Reply:
I believe @jdarwood007 meant that his idea to make an identifierQuery() method would break backward compatibility.